RevenueCatUI/CustomerCenter/Data/CustomerCenterActionWrapper.swift

```diff
@@ -62,57 +62,96 @@ internal enum CustomerCenterInternalAction {
 @MainActor
 final class CustomerCenterActionWrapper {
 
-    // Direct setter closures that will be set by the preference connector
-    var setRestoreStarted: () -> Void = {}
-    var setRestoreFailed: (Error) -> Void = { _ in }
-    var setRestoreCompleted: (CustomerInfo) -> Void = { _ in }
-    var setShowingManageSubscriptions: () -> Void = {}
-    var setRefundRequestStarted: (String) -> Void = { _ in }
-    var setRefundRequestCompleted: (String, RefundRequestStatus) -> Void = { _, _ in }
-    var setFeedbackSurveyCompleted: (String) -> Void = { _ in }
-    var setManagementOptionSelected: (CustomerCenterActionable) -> Void = { _ in }
-    var setPromotionalOfferSuccess: () -> Void = { }
-
-    // The handler for legacy actions
     private let legacyActionHandler: DeprecatedCustomerCenterActionHandler?
 
+    private var restoreStartedCallbacks: [() -> Void] = []
+    private var restoreFailedCallbacks: [(NSError) -> Void] = []
+    private var restoreCompletedCallbacks: [(CustomerInfo) -> Void] = []
+    private var showingManageSubscriptionsCallbacks: [() -> Void] = []
+    private var refundRequestStartedCallbacks: [(String) -> Void] = []
+    private var refundRequestCompletedCallbacks: [(String, RefundRequestStatus) -> Void] = []
+    private var feedbackSurveyCompletedCallbacks: [(String) -> Void] = []
+    private var managementOptionSelectedCallbacks: [(CustomerCenterActionable) -> Void] = []
+    private var promotionalOfferSuccessCallbacks: [() -> Void] = []
```

Comment:
This won't work because there's no way to unregistered the block 🤔 . I think I need to go with another approach.

Reply:
Re-wrote this using combine 👍